src/main/java/org/fogbowcloud/manager/core/OrderStateTransitioner.java

```diff
@@ -110,6 +121,14 @@ private static void doTransition(Order order, OrderState newState) throws Unexpe
         } else {
             if (origin.removeItem(order)) {
                 order.setOrderState(newState);
+
+                // TODO: Updating order in stable storage
+                // (do not update closed orders here since it still have an instance id)
```

Comment:
I did not understand why not updating closed order in the database (I see no one cares about closed resources, but if we do not introduce a bug in case we do update it, it seems better to avoid one more if in the code)

Reply:
I was not updating closed orders here since it already has been updated in `deactivateOrder(Order order)` method, so we could not add an order with same `ID` and `OrderState` twice in the `timestamp` table. However, now we created the`DEACTIVATE` status in which solve this problem.

So, an order that has been closed is stored twice in stable storage: one when the order is deleted (but instanceId is still not null), and another when it is deactivated (when instanceId is null); we need the deactivate state so that the add in the timestamp table won't break.